libse/Subtitle.cs

```diff
@@ -580,5 +580,21 @@ public string GetFastHashCode()
             return sb.ToString().TrimEnd();
         }
 
+        public string GetAllText(bool appendNewLine)
+        {
+            var sb = new StringBuilder();
+            foreach (Paragraph p in Paragraphs)
+            {
+                if (appendNewLine)
+                {
+                    sb.AppendLine(p.Text);
+                }
+                else
+                {
+                    sb.Append(p.Text.TrimEnd() + ' ');

```

Comment:
And perhaps, to be somewhat consistent with `File.ReadAllText` (which preserves line endings),
 `public string GetAllText(bool joinLines = false)`